### PR TITLE
Adds color changing to cooldown circle and a blink animation when done

### PR DESCRIPTION
**Pull request by Tomeno**

### CODE REVIEW

File: Content.Client/UserInterface/CooldownGraphic.cs [hunk 1]
```diff
@@ -8,28 +8,49 @@ namespace Robust.Client.UserInterface.Controls
...
-            var radius = (Math.Min(SizeBox.Height, SizeBox.Width) / 2) * 0.875f; // 28/32 = 0.875 - 2 pixels inwards from the edge
-            var ring_radius = (Math.Min(SizeBox.Height, SizeBox.Width) / 2) * 0.5625f;
+            var effective_fraction = Progress >= 0f ? Progress : 1f; // we want the full circle for the blink
+
+            var segments = (int)Math.Max(2, Math.Ceiling(maxSegments * effective_fraction) + 1); // ensure that we always have at least 3 vertices, we also need one more segment
+            var max = MathHelper.TwoPi * effective_fraction;
+            var outer_radius = (Math.Min(SizeBox.Height, SizeBox.Width) / 2) * 0.875f; // 28/32 = 0.875 - 2 pixels inwards from the edge
+            var inner_radius = (Math.Min(SizeBox.Height, SizeBox.Width) / 2) * 0.5625f; // 18/32 = 0.5625 - 5 pixels thick
 
             Span <Vector2> vertices = stackalloc Vector2[segments * 2];
```
Comment (ComicIronic) [inline Content.Client/UserInterface/CooldownGraphic.cs:29]: Wait this really stack-allocates this whole thing? What is that at maximum, like 1kb of stack?

Comment (Tyler-IN) [reply Content.Client/UserInterface/CooldownGraphic.cs:29]: Yeah 1kB + span's ptr, offset and len (16)

Comment (ComicIronic) [reply Content.Client/UserInterface/CooldownGraphic.cs:29]: Looks like the default stack is 1mB, so it's not as bad as I first thought.

**Comment by ComicIronic** [comment]
:+1:

File: Content.Client/UserInterface/CooldownGraphic.cs (hunk 1 above)
Comment (Tyler-IN) [reply Content.Client/UserInterface/CooldownGraphic.cs:29]: Oh you meant the how big can the stack get...

**Pull request merged**